unity_jobs_simulator/Assets/Scripts/Robot_Controller.cs

```diff
@@ -0,0 +1,230 @@
+        foreach(ArticulationBody meow in _links) // Now we run through each articulation body and add it to the List _joints for those that aren't fixed
+        {
+            if(meow.jointType == ArticulationJointType.RevoluteJoint)
+            {
+                _joints.Add(meow); // This is where we add the joint 'meow' since we know it isn't a fixed joint and can move 
+            }
+            else if(meow.jointType == ArticulationJointType.PrismaticJoint || meow.jointType == ArticulationJointType.SphericalJoint)
+            {
+                Debug.LogError("All the joints must be of type Fixed or Revolute.", meow);
+            }
+        }
+        Debug.Log("We just added " + _joints.Count + " ArticulationBodies to the joints array");
+
+        if(stiffness.Count != _joints.Count || damping.Count != _joints.Count) // Error message if we don't have correct number of inputs
+        {
+            Debug.LogError("Expected " + _joints.Count + " inputs and Got " + stiffness.Count + " and " + damping.Count + " for the stiffness and damping lists.");
+        }
+
+        if(SetAllJointDriveValues) // We set the Stiffness and Damping values to override the preset if told to do so
+        {
+            for(int i=0; i < _joints.Count ; i++) // Runs through all the joints to set each value to the xDrive for each joint
+            {
+                ArticulationDrive drive = _joints[i].xDrive;
+                drive.stiffness = stiffness[i];
+                drive.damping = damping[i];
+                _joints[i].xDrive = drive;
+            }
+        }
+    }
+
+    void Update() //This only updates the target position variable depending on the input we receive for that frame
+    {
+        if(Input.GetKeyDown(KeyCode.R) || Input.GetKeyDown(KeyCode.Alpha0) || Input.GetKeyDown(KeyCode.Keypad0))
+        {
+            TargetJointAngles.Clear();
+            TargetJointAngles.AddRange(_zeroList);
+        }
+        if(Input.GetKeyDown(KeyCode.A))
+        {
+            List<float> testPos1 = new List<float>(){100, -60, 120, 180, -90, 45};
+            TargetJointAngles.Clear();
+            TargetJointAngles.AddRange(testPos1);
+        }
+        if(Input.GetKeyDown(KeyCode.S))
+        {
+            List<float> testPos2 = new List<float>(){-45, -30, 15, 0, 90, 90};
+            TargetJointAngles.Clear();
+            TargetJointAngles.AddRange(testPos2);
+        }
+    }
+
+    void FixedUpdate()
+    {
+        UpdateCurrentDriveTargetAndJointAngles(); // We need to update the current position of the Robot to ensure we know exactly where it is
+        
+        if(SamePos(CurrentJointAngles, TargetJointAngles)) // If we are at the preset target, FixedUpdate shouldn't run so we return nothing
+        {
+            return;
+        }
+
+        if(!(SamePos(_currentDriveTarget, _enroute))) // If the route has changed, we need to restart the motion
```

Comment:
Even when the robot is moving and TargetJointAngles is unchanged, this if condition gets executed. This is because the drive target is constantly being changed when you call AdjustRobotAngles. Should this be comparing _enroute and TargetJointGoals instead? You end up calculating increments constantly as well, as a result of this.

Reply:
Recalculating increments every time is a reasonable solution, by the way, but based on your comments, it doesn't seem like this was the intended behaviour.